proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,1028 @@
+*   HTTP CDN delivery of configuration (list collections with references)
+*   Handcrafted configuration on a filesystem (list collections with inline entries)
+*   Migration of an existing state-of-the-world (SotW) Envoy configuration (list collections with inline entries)
+*   LEDS and high scalability delta APIs (glob collections)
+
+For any other use case with a:
+
+*   Small collection of objects or relatively static configuration;
+    list collections with inline entries are recommended. This is both
+    performant and requires minimal management server state tracking and round
+    trips.
+*   Medium sized collection of objects; list collections with reference entries.
+    The management server only needs to handle subscribe/unsubscribe state
+    tracking, since both the collection and its constituent resources are
+    independent xDS resources.
+*   For any other use case with a large collection of objects and/or highly
+    dynamic updates, glob collections are recommended. The management server needs to
+    provide full delta diffs, since there is no explicit collection object to
+    inform the client when resources come and go.
+
+### Transport network addressing
+
+An `xdstp://` URI specifies a logic authority and it’s necessary to map to some physical 
+network address during the resource fetch process. In the existing xDS configuration
+fetch configuration, this capability is provided by a `ConfigSource` delivered alongside a
+resource name in the xDS resource graph.
+
+This proposal adds to `ConfigSource` the ability to specify the
+authorities for which the `ConfigSource` may be used:
+
+```
+message Authority {
+  string name = 1;
+
+  … when signing is added, items such as CA trust chain, cert pinning … 
+}
+
+message ConfigSource {
+  repeated Authority authorities = N;
+
+  …  extant ConfigSource … 
+}
+```
+
+When the `ConfigSource` adjacent to a resource name matches the URI authority,
+the `ConfigSource` will continue to be used.
+
+When there is no match, the xDS client will fallback to the bootstrap to map
+from authority to `ConfigSource.
+
+
+To support ADS with multiple control planes, the
+[`ApiType`](https://github.com/envoyproxy/envoy/blob/31128e7dc22355876020188bc8feb99304663041/api/envoy/config/core/v3/config_source.proto#L44)
+enum in `ApiConfigSource` will be augmented with `AGGREGATED_DELTA_GRPC`.
+This will replace and deprecate the existing `ads` field in `ConfigSource`. When
+used in `xdstp://` authority resolution, an `ConfigSource` will act as an ADS
+transport if configured with this new enum value. Instead of pointing to a
+single shared ADS `ConfigSource` declared in the bootstrap, any resource
+authority (regardless of type) that maps to a `ConfigSource` with
+`AGGREGATED_DELTA_GRPC` will be multiplexed on the xDS stream specified by the
+`ConfigSource`.
+
+### Discovery request and responses
+
+State-of-the-world (SotW) `Discovery{Request,Response}` are not supported with
```

Comment:
Now that we're using strings instead of structured protos for URIs on the wire, I don't think there's actually any reason to not support xdstp names in SotW.  I think the only thing that doesn't work in SotW is glob collections, because there's no  way for the server to delete a resource from the collection.

Reply:
True, but I rather think about this additively; land the delta spec and add SotW as needed.

Reply:
I'm okay with that, but I do think that we'll need another xRFC for the new naming scheme in SotW in the near future, because gRPC is planning to implement that.  And that xRFC will be very short; it will basically say "everything from the original xRFC except for glob collections works for StoW too".